File: frontend/src/shared/types/analyst.ts
export interface AnalystContext {
  source_type: string;
  title: string;
  content: string;
  relevance: number;
}

export interface AnalystRange {
  from: string;
  to: string;
}

export interface AnalystChatRequest {
  session_id: string;
  user_message: string;
  preferred_range?: AnalystRange;
}

export interface AnalystChatResponse {
  session_id: string;
  answer: string;
  contexts: AnalystContext[];
}

export interface AutoCommentModuleContribution {
  name: string;
  value: number;
}

export interface AutoCommentRequest {
  date: string;
  lsi: number;
  status: string;
  module_contributions: AutoCommentModuleContribution[];
  active_flags: string[];
  upcoming_events: string[];
}

export interface AutoCommentResponse {
  comment: string;
  retrospective: string;
  outlook: string;
}

File: backend/internal/dto/analyst.go
package dto

type GenerateAutoCommentRequest struct {
	Date                string       `json:"date"`
	LSI                 float64      `json:"lsi"`
	Status              string       `json:"status"`
	ModuleContributions []NamedValue `json:"module_contributions"`
	ActiveFlags         []string     `json:"active_flags"`
	UpcomingEvents      []string     `json:"upcoming_events"`
}

type AutoCommentResponse struct {
	Comment       string `json:"comment"`
	Retrospective string `json:"retrospective"`
	Outlook       string `json:"outlook"`
}

type ChatRequest struct {
	SessionID      string     `json:"session_id"`
	UserMessage    string     `json:"user_message"`
	PreferredRange *DateRange `json:"preferred_range,omitempty"`
}

type ChatContext struct {
	SourceType string  `json:"source_type"`
	Title      string  `json:"title"`
	Content    string  `json:"content"`
	Relevance  float64 `json:"relevance"`
}

type ChatResponse struct {
	SessionID string        `json:"session_id"`
	Answer    string        `json:"answer"`
	Contexts  []ChatContext `json:"contexts"`
}
